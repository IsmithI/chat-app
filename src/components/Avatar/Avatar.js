import React from 'react'
import styles from './Avatar.module.scss';

export function Avatar (props) {
	const classes = [styles.avatar];

	if (props.small) classes.push(styles.small)

	return (
		<div className={classes.join(' ')}>
			{props.src && (
				<img src={props.src} alt="avatar"/>
			)}
		</div>
	)
}