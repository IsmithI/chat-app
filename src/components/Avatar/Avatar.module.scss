@import "../../styles/colors";
@import "../../styles/shapes";

.avatar {
	$size: spacing(4);

	min-width: $size;
	max-width: $size;
	min-height: $size;
	max-height: $size;

	box-sizing: border-box;

	display: flex;
	justify-content: center;
	align-content: center;
	align-items: center;

	background-color: $gray;
	border-radius: $radius;

	&.small {
		$size: spacing(2);
		min-width: $size;
		max-width: $size;
		min-height: $size;
		max-height: $size;
		border-radius: $radiusSmall;
	}
}
