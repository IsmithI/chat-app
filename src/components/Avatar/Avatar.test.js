import React from 'react';
import renderer from 'react-test-renderer';
import { Avatar } from './Avatar';

test('Avatar component renders successfully', () => {
	const component = renderer.create(
		<Avatar letter='AB'/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})