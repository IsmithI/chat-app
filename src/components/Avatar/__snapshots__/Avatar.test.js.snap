// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders successfully 1`] = `
<div
  className="avatar"
>
  <span
    className="letter"
  >
    AB
  </span>
</div>
`;
